flake.nix

```diff
@@ -25,6 +25,36 @@
               allowUnfree = true; # needed to get android stuff to compile
             };
           };
+          linux_kernel_upstream = builtins.fetchurl {
+            url = "https://cdn.kernel.org/pub/linux/kernel/v6.x/linux-6.7.6.tar.xz";
+            sha256 = "1lrp7pwnxnqyy8c2l4n4nz997039gbnssrfm8ss8kl3h2c7fr2g4";
+          };
+          # TODO: b/328294742 - This derivation does not currently compile correctly, but it is a starting point.
+          oak_containers_kernel = pkgs.stdenv.mkDerivation {
+            name = "oak_containers_kernel";
+            src = ./oak_containers_kernel;
+            buildInputs = with pkgs; [
+              cowsay
+              curl
+              flex
+              bison
+              bc
+              libelf
+              elfutils
+              glibc
+              glibc.static
+              cpio
+            ];
+            baseInputs = [
+              linux_kernel_upstream
+            ];
+            buildPhase = ''
+              make target/vmlinux
```

Comment:
We want the bzImage, not vmlinux.

Reply:
this is also still unused, but it is a starting point if anyone wants to tweak and try things out.